src/snmalloc/backend_helpers/buddy.h

```diff
@@ -108,7 +112,11 @@ namespace snmalloc
 
       auto bigger = remove_block(size * 2);
       if (bigger == Rep::null)
+      {
+        SNMALLOC_ASSERT(idx == highest_used);
```

Comment:
It seems like we might hit this in OOM scenarios if we've used both halves of the biggest index we've ever used and can't get more?

Reply:
Thanks.  I reworked this a bit, and added some debug invariants.  It is now passing tests, and is easier to follow.